src/lib/components/IndexView.test.tsx

```diff
@@ -0,0 +1,63 @@
+import "@testing-library/jest-dom";
+import { IndexView } from "./IndexView";
+import { describe, it, expect, vi } from "vitest";
+import { render, screen, act, waitFor } from "@testing-library/react";
+import { RouterProvider } from "@tanstack/react-router";
+import { router } from "../../router";
+import { userEvent } from "@testing-library/user-event";
+
+describe("The Index View", () => {
+  it("shows a login form when not logged in", async () => {
+    await act(async () => {
+      render(<RouterProvider router={router} defaultComponent={IndexView} />);
+    });
+
+    expect(screen.getByText(/harry tompson center/i)).toBeInTheDocument();
+    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument();
+    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument();
+    expect(screen.getByText(/log in/i)).toBeInTheDocument();
+  });
+
+  it("shows an error if the user tries to submit an empty form", async () => {
+    const user = userEvent.setup();
+
+    await act(async () => {
+      render(<RouterProvider router={router} defaultComponent={IndexView} />);
+    });
+
+    await user.click(screen.getByRole("button", { name: /log in/i }));
+
+    expect(
+      screen.getByText(/email and password are required/i)
+    ).toBeInTheDocument();
+  });
+
+  it("shows an error message if no fields are filled in and the user hits clicks 'forgot password' button", async () => {
+    const user = userEvent.setup();
+    const originalError = console.error;
```

Comment:
Swap out the real console.error for a mock, just keeps our test output neater when we are testing that things should error.